### PR TITLE
ci(github actions): migrate to github actions

### PR DESCRIPTION
**Pull request by nemni8**
I added 3 workflows, I tested 2 of them (commitlint & test) on this pr.
the 3rd step (publish) will be tested only on master (after pushing a new tag).

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+  schedule:
+    - cron: "*/59 23 * * *"
```
Comment (nemni8) [inline .github/workflows/test.yml:11]: btw I can add cron config to the publish flow and it will release new version every 24 hours. wdyt?

Comment (moshema) [reply .github/workflows/test.yml:11]: As a rule of thumb, I personally dont like GAing when no one's looking. it creates issues in prod that only users will pick up.

Why would we want that cron? to always pick up latest deps?

**Review by moshema — APPROVED** [review]
lgtm

File: .github/workflows/test.yml (hunk 1 above)
Comment (nemni8) [reply .github/workflows/test.yml:11]: there is a task for automating the release of corvid-types on every doc change. using a cron job will achieve the goal of automation releases.

**Pull request merged**